### PR TITLE
Add --gke-subnet-mode flag to GKE deployer in kubetest

### PR DESCRIPTION
**Pull request by tosi3k**

### CODE REVIEW

File: kubetest/gke.go [hunk 1]
```diff
@@ -171,6 +173,15 @@ func newGKE(provider, project, zone, region, network, image, imageFamily, imageP
 		return nil, fmt.Errorf("--gke-shape must include a node pool named 'default', found %q", *gkeShape)
 	}
 
+	switch subnetMode := *gkeSubnetMode; subnetMode {
+	case "auto", "custom":
+		g.subnetMode = subnetMode
+	case "legacy":
```
Comment (mm4tt) [inline kubetest/gke.go:179]: NIT: Let's delete this, saying that we support either 'custom' or 'auto' is enough.

File: kubetest/gke.go [hunk 2]
```diff
@@ -171,6 +173,15 @@ func newGKE(provider, project, zone, region, network, image, imageFamily, imageP
...
+		return nil, fmt.Errorf("--gke-subnet-mode=legacy is illegal since legacy subnet creation mode is deprecated in gcloud, please use 'auto' or 'custom'")
+	default:
+		return nil, fmt.Errorf("--gke-subnet-mode must be set either to 'auto' or 'custom'")
```
Comment (mm4tt) [inline kubetest/gke.go:182]: NIT: Let's add what was provided, e.g. `"..., got: %s", subnetMode)`

File: kubetest/gke.go (hunk 1 above)
Comment (wojtek-t) [reply kubetest/gke.go:179]: +1

Comment (tosi3k) [reply kubetest/gke.go]: Fixed.

File: kubetest/gke.go (hunk 2 above)
Comment (tosi3k) [reply kubetest/gke.go]: Fixed as well.

**Pull request merged**